Add AtkTextInputEventInterface

salanth357 opened the pull request:
This baseclass is used by AtkTextInput to hold a reference to whatever object is currently receiving text events. Leaving it as a placeholder until I can get the vfuncs worked out

Also fills out a bunch of AtkComponentTextInput fields, as well as a couple of the remaining gaps in AtkComponentInputBase based on my initial digging into those vfuncs

wolfcomp commented on FFXIVClientStructs/FFXIV/Component/GUI/AtkTextInput.cs:
```diff
@@ -9,6 +9,7 @@ namespace FFXIVClientStructs.FFXIV.Component.GUI;
 [GenerateInterop]
 [StructLayout(LayoutKind.Explicit, Size = 0xCC0)]
 public unsafe partial struct AtkTextInput {
+    [FieldOffset(0x8)] public AtkTextInputEventInterface* TargetTextInputEventInterface;
```
This is not a straight pointer to `AtkTextInputEventInterface` but a double pointer as it resolved `0x1E0` into `AtkComponentTextInput`

salanth357 replied:
I don't see why it would be? it's not a pointer-to-pointer, it's a pointer-to-AtkTextInputInterface, which happens to be embedded into `AtkComponentTextInput` as a secondary base class, with a special vtbl

wolfcomp reviewed (changes requested):
You will need to double check everything at this point that it's correct for what memory maps to where and what is correctly being decompiled with IDA and Ghidra.

salanth357 commented on FFXIVClientStructs/FFXIV/Component/GUI/AtkComponentInputBase.cs:
```diff
@@ -12,6 +12,8 @@ namespace FFXIVClientStructs.FFXIV.Component.GUI;
 public unsafe partial struct AtkComponentInputBase {
     [FieldOffset(0xC8)] public AtkTextNode* AtkTextNode;
     [FieldOffset(0xD0)] public AtkResNode* CursorContainer;
+    [FieldOffset(0xD8)] public AtkEventManager* EventManager;
```
Hm, you're right- I don't know why I thought there was an `AtkEventManager*` there, I'll recheck all the fields I added to make sure none of the others are wrong

salanth357 commented on FFXIVClientStructs/FFXIV/Component/GUI/AtkComponentTextInput.cs:
```diff
@@ -26,9 +26,25 @@ public unsafe partial struct AtkComponentTextInput : ICreatable {
-    [FieldOffset(0x450)] public Utf8String UnkText04;
-    [FieldOffset(0x4B8)] public Utf8String UnkText05;
+
+    [FieldOffset(0x350)] [Obsolete("Renamed to AvailableLines")] public Utf8String UnkText03;
+    [FieldOffset(0x450)] [Obsolete("Renamed to HighlightedAutoTranslateOptionColorPrefix")] public Utf8String UnkText04;
+    [FieldOffset(0x4B8)] [Obsolete("Renamed to HighlightedAutoTranslateOptionColorSuffix")] public Utf8String UnkText05;
+
+    [FieldOffset(0x350)] public Utf8String AvailableLines;
+    // Utf8Strings containing color macros that are wrapped around the highlighted AutoTranslate option
+    [FieldOffset(0x450)] public Utf8String HighlightedAutoTranslateOptionColorPrefix;
+    [FieldOffset(0x4B8)] public Utf8String HighlightedAutoTranslateOptionColorSuffix;
+
+    [FieldOffset(0x3B8)] public AtkTextNode* AvailableLinesTextNode;
+    [FieldOffset(0x3C0)] public AtkTextNode* AvailableCharsTextNode;
+    [FieldOffset(0x3C8)] public AtkUnitBase* ContainingAddon2; // For whatever reason, the text input _also_ has this
+    [FieldOffset(0x3D0)] public AtkResNode* AutoTranslateMenuNode;
+    [FieldOffset(0x3D8)] public int CompletionOffset; // Offset into the total number of completion items- used for drawing the label
+
+    [FieldOffset(0x3E0), FixedSizeArray] internal FixedSizeArray9<Pointer<AtkComponentButton>> _autoTranslateMenuButtons;
+    [FieldOffset(0x428)] public AtkTextNode* AutoTranslateMenuPageInfoTextNode;
+    [FieldOffset(0x430)] public AtkNineGridNode* AutoTranslateMenuBackground;
```
Will do- I don't remember why I grouped them in this way

salanth357 commented:
Updated the PR- I removed the EventManager (still don't know why I thought that was there...), and validated that the other fields added match what's present in memory in game.
